### PR TITLE
[#1976] Never let a start or end time be greater than duration, or less ...

### PR DESCRIPTION
**Pull request by ScottDowne**
...than 0.

### CODE REVIEW

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @cadecairos received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/2763aaf67d3af47/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @cadecairos received. Current queue size: 1

Live output at: http://botio.butyr.org:8000/95a6278dce50165/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/2763aaf67d3af47/output.txt

Total script time: 1.39 mins

Server is running on http://pr1034.butyr.org/templates/basic/
Run the Test Suite: http://pr1034.butyr.org/test/

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://botio.butyr.org:8000/95a6278dce50165/output.txt

Total script time: 1.61 mins

File: src/core/trackevent.js [hunk 1]
```diff
@@ -164,10 +173,10 @@ define( [
         }
       }
       
-      if( newStart ){
+      if( newStart != null ){
         _popcornOptions.start = newStart;
       }
-      if( newEnd ){
+      if( newEnd != null){
```
Comment (dseif) [inline src/core/trackevent.js]: why do we need to check `newStart` and `newEnd` against null?

File: src/core/trackevent.js [hunk 2]
```diff
@@ -131,13 +132,21 @@ define( [
...
+
       if ( _track && _track._media ) {
         media = _track._media;
-        if( media.ready &&
-            ( newStart > media.duration ||
-            newEnd > media.duration ||
-            newStart < 0 ) ) {
-          throw new TrackEventUpdateException( "invalid-times", "[start] or [end] are not within the duration of media" );
+        duration = media.duration
+        if( media.ready ) {
```
Comment (dseif) [inline src/core/trackevent.js]: `if(` -> `if (`

File: src/core/trackevent.js [hunk 3]
```diff
@@ -131,13 +132,21 @@ define( [
...
+          if ( newStart < 0 ) {
+            newStart = 0;
+          } else if ( newStart > duration ) {
+            newStart = duration;
+          }
+          if ( newEnd < 0 ) {
+            newEnd = 0;
+          } else if ( newEnd > duration ) {
+            newEnd = duration;
+          }
```
Comment (dseif) [inline src/core/trackevent.js:149]: These if blocks can be replaced by something like:

`newStart = Math.min( Math.max( 0, newStart ), duration );`

the same goes for `newEnd`

Comment (cadecairos) [inline src/core/trackevent.js:149]: might be faster to stick with the if/else: http://jsperf.com/ifelsevsmax

Comment (dseif) [inline src/core/trackevent.js:149]: True! Looks like Scott's approach is better.

File: src/core/trackevent.js (hunk 1 above)
Comment (ScottDowne) [inline src/core/trackevent.js]: This means 0 resolves as true. This is only false if it is null or undefined.

Comment (dseif) [inline src/core/trackevent.js]: ah true, other than the style nit then this is good to go

**Pull request closed without merging**